types/part_set.go

```diff
@@ -299,6 +299,19 @@ func (ps *PartSet) AddPart(part *Part) (bool, error) {
 		return false, nil
 	}
 
+	ps.mtx.Lock()
+	defer ps.mtx.Unlock()
```

Comment:
is this the reason why we're copying parts of the addPartWithoutProof in this method?

Reply:
It's reverting `AddPart` to match upstream and duplicating `AddPart` to `AddPartWithoutProof` and stripping the proof logic from that method.